### DISCUSSION TITLE
On measuring half light radius for system with single sersic profile

### DISCUSSION THREAD

**Discussion by mh-tanver** [AstroBridge/BDLensing#256] [Q&A] [answered]
I am confused about measuring half light radius using the same procedures as systems with double sersics. How does the **R_eff** differ from the **R_sersic** in the context of single Sersic profile systems?

**Comment by SulymanHossain** [comment]
@ajshajib

**Comment by ajshajib** [comment, marked as answer]
For a single Sersic profile, $R_{\rm Sersic}$ is the same as $R_{\rm eff}$.

**Reply by SulymanHossain** [reply]
While extracting the flux value of the Sersic profiles through LightProfileAnalysis.flux_components(), I'm getting a half-light radius for every single Sersic, and it's different compared to the corresponding R_sersic. What does this half-light radius for a single profile mean?

**Reply by ajshajib** [reply]
I suggest you check the code to see how it's being computed. That may give you some clarification on your question.

**Reply by SulymanHossain** [reply]
I got confused because it's using the same half_light_radius() function. The only difference I could find is that the flux_components() function doesn't use centroids of the profiles while calculating half-light radius.   
```python
def flux_components(self, kwargs_light, grid_num=400, grid_spacing=0.01):
        """Computes the total flux in each component of the model.

        :param kwargs_light:
        :param grid_num:
        :param grid_spacing:
        :return:
        """
        flux_list = []
        R_h_list = []
        x_grid, y_grid = util.make_grid(numPix=grid_num, deltapix=grid_spacing)
        kwargs_copy = copy.deepcopy(kwargs_light)
        for k, kwargs in enumerate(kwargs_light):
            if "center_x" in kwargs_copy[k]:
                kwargs_copy[k]["center_x"] = 0
                kwargs_copy[k]["center_y"] = 0
            light = self._light_model.surface_brightness(
                x_grid, y_grid, kwargs_copy, k=k
            )
            flux = np.sum(light) * grid_spacing**2
            R_h = analysis_util.half_light_radius(light, x_grid, y_grid)
            flux_list.append(flux)
            R_h_list.append(R_h)
        return flux_list, R_h_list
```

**Reply by ajshajib** [reply]
Sersic profile is defined such that an ellipse with intermediate axis length $R_{\rm sersic}$ will contain half the total light. The function here finds the radius of the circle that contains half the light. So, some differences are inevitable from the difference in definition, and more differences will arise from the numerical precision (`grid_spacing` and `grid_num`). However, the two values should be pretty close for high numerical precision.

**Reply by SulymanHossain** [reply]
Understood. Thanks for the explanation. For a Sersic profile, I'm getting half-light radius = 1.85 while R_sersic is 1.916 with `grid_spacing = 0.01` and  `grid_num = 10 / 0.01`. I don't know whether it's close enough.

**Comment by rafee-adnan** [comment]
@JobairHasan
